transforms/update-react-imports.js

```diff
@@ -54,16 +80,236 @@ module.exports = function(file, api, options) {
       })
       .size() > 0;
 
-  const reactPath = reactImportPath.paths()[0];
-  const specifiers = reactPath.value.specifiers;
+  // local: imported
+  const reactIdentifiers = {};
+  const reactTypeIdentifiers = {};
+  let canDestructureReactVariable;
+  if (reactImportInUse && isDefaultImport) {
+    // Checks to see if the react variable is used itself (rather than used to access its properties)
+    canDestructureReactVariable =
+      root
+        .find(j.Identifier, {
+          name: 'React',
+        })
+        .filter(path => {
+          return path.parent.parent.value.type !== 'ImportDeclaration';
+        })
+        .filter(
+          path =>
+            !(
+              path.parent.value.type === 'MemberExpression' &&
+              path.parent.value.object.name === 'React'
+            ) &&
+            !(
+              path.parent.value.type === 'QualifiedTypeIdentifier' &&
+              path.parent.value.qualification.name === 'React'
+            ) &&
+            !(
+              path.parent.value.type === 'JSXMemberExpression' &&
+              path.parent.value.object.name === 'React'
+            ),
+        )
+        .size() === 0;
 
-  for (let i = 0; i < specifiers.length; i++) {
-    const specifier = specifiers[i];
-    if (specifier.local.name === 'React') {
-      if (specifier.type === 'ImportNamespaceSpecifier' && !keepReactImport) {
-        j(reactPath).remove();
+    if (canDestructureReactVariable) {
+      // If a variable that we are trying to import has already been declared,
+      // do not try to destructure imports
+      root
+        .find(j.QualifiedTypeIdentifier, {
+          qualification: {
+            type: 'Identifier',
+            name: 'React',
+          },
+        })
+        .forEach(path => {
+          const id = path.value.id.name;
+          // This is a typeof import so it isn't actually a type
+          if (path.parent.parent.value.type === 'TypeofTypeAnnotation') {
```

Comment:
Move the comment inside the condition 
```suggestion
          if (path.parent.parent.value.type === 'TypeofTypeAnnotation') {
            // This is a typeof import so it isn't actually a type
```